ldap/servers/slapd/conntable.c

```diff
@@ -471,19 +502,32 @@ connection_table_move_connection_on_to_active_list(Connection_Table *ct, Connect
     connection_table_dump_active_connection(c);
 #endif
 
-    c->c_next = ct->c[0].c_next;
+    /* Map new connection to connection table list */
+    c->c_ct_list = connection_table_get_list(ct);
```

Comment:
listener thread is assigned here when the connection is active. Shouldn't we reset c_ct_list to some specific value during connection_table_move_connection_out_of_active_list

Reply:
I agree it does not need a reset but I am a bit concern that a connection on the freelist keeps its previous listener assigment. I would prefere a setting like: free => c->ct_list == -1, active => c->ct_list = [0-3]

Reply:
Sure, I can do that.